internal/shared/case_status.go

```diff
@@ -0,0 +1,168 @@
+	CaseStatusTypeDeRegistered
+	CaseStatusTypeSuspended
+)
+
+var caseStatusTypeMap = map[string]CaseStatus{
+	"draft":                    CaseStatusTypeDraft,
+	"Draft":                    CaseStatusTypeDraft,
+	"In progress":              CaseStatusTypeInProgress,
+	"in-progress":              CaseStatusTypeInProgress,
+	"Statutory waiting period": CaseStatusTypeStatutoryWaitingPeriod,
+	"statutory-waiting-period": CaseStatusTypeStatutoryWaitingPeriod,
+	"Do not register":          CaseStatusTypeDoNotRegister,
+	"do-not-register":          CaseStatusTypeDoNotRegister,
+	"Expired":                  CaseStatusTypeExpired,
+	"expired":                  CaseStatusTypeExpired,
+	"Registered":               CaseStatusTypeRegistered,
+	"registered":               CaseStatusTypeRegistered,
+	"Cannot register":          CaseStatusTypeCannotRegister,
+	"cannot-register":          CaseStatusTypeCannotRegister,
+	"Cancelled":                CaseStatusTypeCancelled,
+	"cancelled":                CaseStatusTypeCancelled,
+	"De-registered":            CaseStatusTypeDeRegistered,
+	"de-registered":            CaseStatusTypeDeRegistered,
+	"Suspended":                CaseStatusTypeSuspended,
+	"suspended":                CaseStatusTypeSuspended,
+	"Perfect":                  CaseStatusTypePerfect,
+	"Pending":                  CaseStatusTypePending,
+	"Payment Pending":          CaseStatusTypePaymentPending,
+	"Reduced Fees Pending":     CaseStatusTypeReducedFeesPending,
+	"Rejected":                 CaseStatusTypeRejected,
+	"Withdrawn":                CaseStatusTypeWithdrawn,
+	"Return - unpaid":          CaseStatusTypeReturnUnpaid,
+	"Deleted":                  CaseStatusTypeDeleted,
+	"Revoked":                  CaseStatusTypeRevoked,
+}
+
+func (cs CaseStatus) String() string {
+	switch cs {
+	case CaseStatusTypeDraft:
+		return "Draft"
+	case CaseStatusTypeInProgress:
+		return "In progress"
+	case CaseStatusTypeStatutoryWaitingPeriod:
+		return "Statutory waiting period"
+	case CaseStatusTypeDoNotRegister:
+		return "Do not register"
+	case CaseStatusTypeExpired:
+		return "Expired"
+	case CaseStatusTypeRegistered:
+		return "Registered"
+	case CaseStatusTypeCannotRegister:
+		return "Cannot register"
+	case CaseStatusTypeCancelled:
+		return "Cancelled"
+	case CaseStatusTypeDeRegistered:
+		return "De-registered"
+	case CaseStatusTypeSuspended:
+		return "Suspended"
+	case CaseStatusTypePerfect:
+		return "Perfect"
+	case CaseStatusTypePending:
+		return "Pending"
+	case CaseStatusTypePaymentPending:
+		return "Payment Pending"
+	case CaseStatusTypeReducedFeesPending:
+		return "Reduced Fees Pending"
+	case CaseStatusTypeRejected:
+		return "Rejected"
+	case CaseStatusTypeWithdrawn:
+		return "Withdrawn"
+	case CaseStatusTypeReturnUnpaid:
+		return "Return - unpaid"
+	case CaseStatusTypeDeleted:
+		return "Deleted"
+	case CaseStatusTypeRevoked:
+		return "Revoked"
+	default:
+		return ""
+	}
+}
+
+func (cs CaseStatus) CaseStatusColour() string {
```

Comment:
Don't think we need to include "CaseStatus" in the method name because that's known from the type?
```suggestion
func (cs CaseStatus) Colour() string {
```